content/400-pulse/400-api-reference.mdx

```diff
@@ -19,226 +17,438 @@ model User {
 
-Pulse returns the values of the changed record as an object named `deleted`. This represents the state of the record _before_ the change event.
+<TabbedContent code>
+<TabItem value="Type">
 
-##### Example
+```ts
+PulseSubscription<
+  | PulseCreateEvent<{
+      id: number;
+      name: string | null;
+      email: string;
+    }>
+  | PulseUpdateEvent<{
+      id: number;
+      name: string | null;
+      email: string;
+    }>
+  | PulseDeleteEvent<{
+      id: number;
+      name: string | null;
+      email: string;
+    }>
+>;
+```
+
+</TabItem>
+<TabItem value="No filter arguments">
+```ts
+const subscription = await prisma.user.subscribe()
+```
+
+</TabItem>
+</TabbedContent>
+
+Depending on the arguments you provide, the return type may change. For example, if you filter for only `create` events, the type will get adjusted:
 
-<CodeWithResult outputResultText="event">
-<cmd>
+<TabbedContent code>
+<TabItem value="Type">
 
+```ts
+PulseSubscription<
+  PulseCreateEvent<{
+    id: number;
+    email: string;
+    name: string | null;
+  }>
+>;
+```
+
+</TabItem>
+<TabItem value="Filter for `create` events">
 ```ts
 const subscription = await prisma.user.subscribe({
-  delete: {},
+  create: {}
 })
 ```
 
-</cmd>
-<cmdResult>
+</TabItem>
+</TabbedContent>
 
-```ts no-copy
-{
-  action: 'delete',
-  deleted: {
-    id: 1,
-    email: '',
-    name: null
-  }
-}
+### Examples
+
+#### Filter for new `User` records with a non-null value for `name`
+
+```ts
+const subscription = await prisma.user.subscribe({
+  create: {
+    name: { not: null },
+  },
+});
 ```
 
-</cmdResult>
-</CodeWithResult>
+#### Filter for updated `User` records where the `email` ends with `@prisma.io` after the update
 
-<Admonition type="warning">
+```ts
+const subscription = await prisma.user.subscribe({
+  update: {
+    after: {
+      email: { endsWith: "@prisma.io" },
+    },
+  },
+});
+```
 
-By default, the response for `delete` events will only return the primary key of the record. To get the _before_ values of all fields in the record, you must set `REPLICA IDENTITY` to `FULL` on the table(s) you want to get field values for. If this is not configured, defining a filter for deletes will only be possible on the primary key. <br/><br/>
-For example, running the following SQL command will set the `REPLICA IDENTITY` to `FULL` on a table named `User`:
+#### Filter for deleted `User` records where the `email` contains `hello` after the update
 
-```sql
-ALTER TABLE public."User" REPLICA IDENTITY FULL;
+```ts
+const subscription = await prisma.user.subscribe({
+  delete: {
+    email: { contains: "hello" },
+  },
+});
 ```
 
-</Admonition>
+## `stream()` vs `subscribe()`
+
+For the majority of use cases, `stream()` is the recommended option because can provide guarantees for events to arrive on the consumer-side. Note though that because `stream()` requires event persistence to be enabled, this has implications for [pricing](https://www.prisma.io/pricing).
```

Comment:
```suggestion
For the majority of use cases, `stream()` is the recommended option because it can provide guarantees for events to arrive on the consumer-side. Note though that because `stream()` requires event persistence to be enabled, this has implications for [event storage & costs](https://www.prisma.io/pricing).
```

For everyone else than non-existing Pulse users, this new change is a non-change, as they wouldn't have known our previous pricing. I don't want to scare them away that they're using something "too expensive".